python/langsmith/client.py

```diff
@@ -491,6 +504,20 @@ def __init__(
         # Create a session and register a finalizer to close it
         session_ = session if session else requests.Session()
         self.session = session_
+        self.compress_traces = ls_utils.get_env_var("USE_RUN_COMPRESSION")
+        if self.compress_traces:
+            self._futures: set[cf.Future] = set()
+            self._boundary = BOUNDARY
+            self.compressed_runs_buffer: Optional[io.BytesIO] = io.BytesIO()
+            self.compressor_writer: zstd.ZstdCompressionWriter = zstd.ZstdCompressor(
+                level=3, threads=-1
+            ).stream_writer(self.compressed_runs_buffer, closefd=False)
+            self._buffer_lock: threading.Lock = threading.Lock()
+            self._data_available_event = threading.Event()
+            self._run_count: int = 0
```

Comment:
I'd prefer if we did not track statefulness as a separate property if we can help it

Reply:
agreed, fixed